tests/utils.py

```diff
@@ -252,9 +253,129 @@ def calculate_accuracy(original_outputs, compiled_outputs):
             output_pccs
         ), f"No comparable outputs:\noriginal_outputs:\n{original_outputs}\ncompiled_outputs:\n{compiled_outputs}"
         accuracy = torch.mean(torch.tensor(output_pccs)).item()
+    elif (isinstance(original_outputs, list) and isinstance(compiled_outputs, list)) or (
+        isinstance(original_outputs, tuple) and isinstance(compiled_outputs, tuple)
+    ):
+        # Handle case where outputs are lists
+        output_pccs = []
+        assert len(original_outputs) == len(compiled_outputs), (
+            f"Original and compiled output do not have the same length."
+            f"original length: {len(original_outputs)}\ncompiled length: {len(compiled_outputs)}"
+        )
+        for original_outputs, compiled_outputs in zip(original_outputs, compiled_outputs):
+            if isinstance(original_outputs, torch.Tensor) and isinstance(compiled_outputs, torch.Tensor):
+                _, pcc = comp_pcc(original_outputs, compiled_outputs)
+                output_pccs.append(pcc)
+        assert (
+            output_pccs
+        ), f"No comparable outputs:\noriginal_outputs:\n{original_outputs}\ncompiled_outputs:\n{compiled_outputs}"
+        accuracy = torch.mean(torch.tensor(output_pccs)).item()
     elif isinstance(original_outputs, torch.Tensor) and isinstance(compiled_outputs, torch.Tensor):
         # Handle case where outputs are Pytorch Tensors
         _, accuracy = comp_pcc(original_outputs, compiled_outputs)
     else:
         accuracy = None
     return accuracy
+
+
+def get_input_vals_from_metric_str(op_name, input_strings):
```

Comment:
this method itself requires good testing

Reply:
Ok, Can I add the testing after this PR merged?